lib/backend/stat_cache_proxy.dart

```diff
@@ -2,18 +2,19 @@ import 'package:files/backend/database/model.dart';
 import 'package:files/backend/providers.dart';
 
 class StatCacheProxy {
-  final Map<String, EntityStat> _runtimeCache = {};
-
   StatCacheProxy();
+  final Map<String, EntityStat> _runtimeCache = {};
 
   Future<EntityStat> get(String path) async {
     if (!_runtimeCache.containsKey(path)) {
-      final stat = await helper.get(path)
-        ..fetchUpdate();
+      final stat = await helper.get(path);
+      await stat.fetchUpdate();
       _runtimeCache[path] = stat;
       return stat;
     }
 
+    // TODO(@HrX03): is this correct?
+    // ignore: unawaited_futures
     return _runtimeCache[path]!..fetchUpdate();
```

Comment:
doesn't seem so, the future will need to be awaited by the caller, maybe tho it means the fetchUpdate call is not awaited? in that case i'd just separate the two things and make a var + call + return. tho it seems like we wanna return fetchUpdate as it's the only future present here

Reply:
fixed